### PR TITLE
Build: Switch from JSHint to ESLint with the jquery preset

### PR DESCRIPTION
**Pull request by mgol**
Also, fix linting violations in the process

### CODE REVIEW

**Comment by raphamorim** [comment]
LGTM 👍

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+	"extends": "eslint-config-jquery",
+	"root": true,
+	"rules": {
+
+		// Increase max-len to 120 for now, there are too many violations and the rule
+		// is not auto-fixable.
```
Comment (leobalter) [inline .eslintrc.json]: I'm ok to remove this later, but this comment is not necessary and it's invalid for the syntax.

File: Gruntfile.js [hunk 2]
```diff
@@ -96,13 +89,22 @@ grunt.initConfig({
 		}
 	},
 
-	jshint: {
+	eslint: {
 		options: {
-			jshintrc: true
+
+			// See https://github.com/sindresorhus/grunt-eslint/issues/119
```
Comment (leobalter) [inline Gruntfile.js]: 404, no issues tracker available on https://github.com/sindresorhus/grunt-eslint

File: jquery.color.js [hunk 3]
```diff
@@ -147,26 +151,27 @@ each( spaces, function( spaceName, space ) {
...
+
 		// we add mod before modding to make sure that negatives values
 		// get converted properly: -10 -> 350
-		return (value + type.mod) % type.mod;
+		return ( value + type.mod ) % type.mod;
 	}
 
 	// for now all property types without mod have min and max
-	return 0 > value ? 0 : type.max < value ? type.max : value;
+    return Math.min( type.max, Math.max( 0, value ) );
```
Comment (leobalter) [inline jquery.color.js]: ???

Apparently eslint fixed this in a wrong way.

File: .eslintrc.json (hunk 1 above)
Comment (mgol) [inline .eslintrc.json]: It is valid. This type of ESLint config allows comments, similarly to how .jshintrc allowed comments.

File: Gruntfile.js (hunk 2 above)
Comment (mgol) [inline Gruntfile.js]: Sindre was getting too many issues reported and he disabled the bug tracker... He's don it before and restored later so this issue is most likely still there. Do you want me to remove this comment? Maybe there's a version in Web Archive?

File: jquery.color.js (hunk 3 above)
Comment (mgol) [inline jquery.color.js]: This wasn't ESLint, it was me. What's wrong here?

Comment (leobalter) [inline jquery.color.js]: indentation using spaces instead of tab.

Comment (mgol) [inline jquery.color.js]: Ah, true. We don't enforce indentation via ESLint yet, it's waiting for https://github.com/eslint/eslint/issues/6690 to get fixed. I applied the change.

**Comment by mgol** [comment]
@leobalter I applied indentation fixes & replied to your other comments. PTAL.

**Pull request merged**